cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -370,6 +370,53 @@ typedef enum {
     STM32_LPUART,           /**< STM32 Low-power UART (LPUART) module type */
 } uart_type_t;
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+/* Invalid mask forces enums to be 32 bits */
+#define UART_INVALID_MODE 0x8000000
```

Comment:
> why would you want to force the enum to be 32 bits?

This is because data_bits_t is used as an argument, if a certain type cpu is used (that doesn't contain the M1 bit) like the stm32f103rb then the size is 8 bits (more likely compiler dependent).  Since the data_bits variable is being used and altered when the mode is set depending on parity it must be a uint32_t type for the `data_bits = USART_CR1_M;` call.
The other way that was implemented was having a holder but I think this is a bit cleaner

Documentation should be updated though!

Reply:
Done